### PR TITLE
Add support for gitlab templates

### LINKED ISSUE #128: Add support for GitLab templates

### PR DESCRIPTION
**Pull request by eranelbaz**
### Issue & Steps to Reproduce / Feature Request
resolves #128

Allow the user to create gitlab templates
### Solution
Add terraform attributes `gitlab_token_id` and `gitlab_project_id`
calculate `isGitlab` using those attributes

### CODE REVIEW

File: client/model.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ type TemplateCreatePayload struct {
 	TokenName            string           `json:"tokenName"`
 	TokenId              string           `json:"tokenId"`
```
Comment (razbensimon) [inline client/model.go:109]: add omit to TokenId also here

File: env0/data_template.go [hunk 2]
```diff
@@ -79,6 +79,16 @@ func dataTemplate() *schema.Resource {
 				Description: "The env0 application installation id on the relevant github repository",
 				Optional:    true,
 			},
+			"token_id": {
+				Type:        schema.TypeString,
+				Description: "The env0 application token id on the relevant gitlab account",
```
Comment (razbensimon) [inline env0/data_template.go:84]: the token_id is used also for Other vcs, so change the description accordingly

File: env0/data_template.go [hunk 3]
```diff
@@ -131,6 +141,11 @@ func dataTemplateRead(ctx context.Context, d *schema.ResourceData, meta interfac
 		d.Set("github_installation_id", template.GithubInstallationId)
 	}
 
+	if template.TokenId != "" {
+		d.Set("token_id", template.TokenId)
+		d.Set("gitlab_project_id", template.GitlabProjectId)
+	}
```
Comment (razbensimon) [inline env0/data_template.go:147]: here also, we dont need to set GitlabProjectId if it is just Other VCS.
so probably need to do something like this:

```go
if template.TokenId != "" {
	d.Set("token_id", template.TokenId)
        if template.GitlabProjectId != 0 {
        	d.Set("gitlab_project_id", template.GitlabProjectId)
        }
}
```

File: env0/resource_template.go [hunk 4]
```diff
@@ -99,6 +99,16 @@ func resourceTemplate() *schema.Resource {
 				Description: "The env0 application installation id on the relevant github repository",
 				Optional:    true,
 			},
+			"token_id": {
+				Type:        schema.TypeString,
+				Description: "The env0 application token id on the relevant gitlab account",
```
Comment (razbensimon) [inline env0/resource_template.go:104]: here again

File: env0/resource_template.go [hunk 5]
```diff
@@ -120,6 +130,19 @@ func templateCreatePayloadFromParameters(d *schema.ResourceData) (client.Templat
 	if githubInstallationId, ok := d.GetOk("github_installation_id"); ok {
 		result.GithubInstallationId = githubInstallationId.(int)
 	}
+	if tokenId, ok := d.GetOk("token_id"); ok {
+		result.TokenId = tokenId.(string)
+	}
+	if gitlabProjectId, ok := d.GetOk("gitlab_project_id"); ok {
+		result.GitlabProjectId = gitlabProjectId.(int)
+	}
+
```
Comment (razbensimon) [inline env0/resource_template.go:139]: does it work? i remember we do not save this on db

**Review by razbensimon — DISMISSED** [review]
Nice work. just need to use this version and QA that it is working

File: env0/data_template.go (hunk 2 above)
Comment (eranelbaz) [reply env0/data_template.go:84]: WDYM?
Where we can use that for Other vcs?
can you suggest different description?

File: env0/resource_template.go (hunk 4 above)
Comment (eranelbaz) [reply env0/resource_template.go:104]: Here again

File: env0/resource_template.go (hunk 5 above)
Comment (eranelbaz) [reply env0/resource_template.go]: 5a4bc914

File: env0/data_template.go (hunk 3 above)
Comment (eranelbaz) [reply env0/data_template.go]: decc5707

File: client/model.go (hunk 1 above)
Comment (eranelbaz) [reply client/model.go]: fc896435

**Comment by eranelbaz** [comment]
7b8fc61a fixed CI

File: env0/resource_template.go (hunk 5 above)
Comment (razbensimon) [reply env0/resource_template.go:139]: I didnt thought you need to delete it, just tell me if it works or not - or if we SHOULD make it work?

File: env0/data_template.go (hunk 3 above)
Comment (razbensimon) [reply env0/data_template.go]: why outside the first `if`?
we should not Set it if there is no tokenID first

File: env0/data_template.go (hunk 2 above)
Comment (razbensimon) [reply env0/data_template.go:84]: something like:
`the token id used for private git repos or for integration with GitLab. More info: https://docs.env0.com/docs/templates#manage-git-connectivity`

Your call

File: env0/resource_template.go (hunk 5 above)
Comment (eranelbaz) [reply env0/resource_template.go:139]: I don't think we should
I saw on our FE we don't get it so i guess its not returned from BE

File: env0/data_template.go (hunk 3 above)
Comment (eranelbaz) [reply env0/data_template.go]: it doesn't really matter in my taste,
WDYT?

File: env0/data_template.go (hunk 2 above)
Comment (eranelbaz) [reply env0/data_template.go:84]: 1e8ebd59

File: env0/data_template.go (hunk 3 above)
Comment (eranelbaz) [reply env0/data_template.go]: 52220487
as we talked added to resource

**Review by razbensimon — APPROVED** [review]
nice!

**Pull request merged**